java/algorithm/src/main/java/org/apache/arrow/algorithm/deduplicate/DeduplicationUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.deduplicate;
+
+import org.apache.arrow.util.DataSizeRoundingUtil;
+import org.apache.arrow.util.Preconditions;
+import org.apache.arrow.vector.BitVectorHelper;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.ValueVector;
+
+import io.netty.buffer.ArrowBuf;
+
+/**
+ * Utilities for vector deduplication.
+ */
+class DeduplicationUtils {
+
+  /**
+   * Gets the start positions of the first distinct values in a vector.
+   * @param vector the target vector.
+   * @param runStarts the bit set to hold the start positions.
+   * @param <V> vector type.
+   */
+  public static <V extends ValueVector> void populateRunStartIndicators(V vector, ArrowBuf runStarts) {
+    int bufSize = DataSizeRoundingUtil.divideBy8Ceil(vector.getValueCount());
+    Preconditions.checkArgument(runStarts.capacity() >= bufSize);
+    runStarts.setZero(0, bufSize);
+
+    BitVectorHelper.setValidityBitToOne(runStarts, 0);
+
+    for (int i = 1; i < vector.getValueCount(); i++) {
+      if (!vector.equals(i, vector, i - 1)) {
```

Comment:
can you rebase, I think this API has been eliminated?

Reply:
@emkornfield Thank you for the kind reminder. This is blocked by another issue: ARROW-6266. Let's fix that one first.

Reply:
@emkornfield Rebased. Please take a look. Thank you.